### PR TITLE
docs(checkbox): A11y checkbox docs

### PR DESCRIPTION
**Pull request by aledavila**
ref #491 

#### Changelog

**New**

- adds Accessibility docs to checkbox

content should match https://pages.github.ibm.com/IBMa/Carbon_Integration/Doc/Checkbox-doc.html

### CODE REVIEW

**Comment by connor-leech** [comment]
@aledavila Can we switch the headers to sentence case? I can leave it in comments but that would be a pain for both of us

**Comment by connor-leech** [comment]
@aledavila Looks like I can commit directly to the branch, I updated it.

**Comment by aledavila** [comment]
@connor-leech thanks

File: src/pages/components/checkbox/accessibility.mdx [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+          <StructuredListCell head>Environment </StructuredListCell>
+          <StructuredListCell head>Results</StructuredListCell>
+        </StructuredListRow>
+      </StructuredListHead>
+      <StructuredListBody>
+        <StructuredListRow>
+          <StructuredListCell>
+            - macOS Mojave version 10.14.6 with VoiceOver
+            <br />
+            - Chrome Version Version 77.0.3865.90
```
Comment (connor-leech) [inline src/pages/components/checkbox/accessibility.mdx]:
```suggestion
            - Chrome version 77.0.3865.90
```

Comment (aledavila) [reply src/pages/components/checkbox/accessibility.mdx]: This is what happens when you copy paste 😨

File: src/pages/components/checkbox/accessibility.mdx [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+<AnchorLinks>
+  <AnchorLink>How it works</AnchorLink>
+  <AnchorLink>Accessibility considerations</AnchorLink>
+  <AnchorLink>Resources</AnchorLink>
+  <AnchorLink>Accessibility testing</AnchorLink>
+</AnchorLinks>
+
+## How it works
+
+TThe Checkbox Component is used to provide a list of options where the user can select multiple options, including all or none. A checkbox control has three possible states indicated by the value of its aria-checked attribute, “true” when selected, “false” when unselected and “mixed” when in the indeterminate state. The indeterminate state comes into play when the checkbox contains a sublist of selections, some of which are selected, and some unselected.
```
Comment (connor-leech) [inline src/pages/components/checkbox/accessibility.mdx]:
```suggestion
The Checkbox Component is used to provide a list of options where the user can select multiple options, including all or none. A checkbox control has three possible states indicated by the value of its aria-checked attribute, “true” when selected, “false” when unselected and “mixed” when in the indeterminate state. The indeterminate state comes into play when the checkbox contains a sublist of selections, some of which are selected, and some unselected.
```

**Pull request merged**